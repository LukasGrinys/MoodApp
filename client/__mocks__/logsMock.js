export const testLogs = [
    {
        id: '001',
        rating: 10,
        date: "TEST-DATE",
        text: "Test text"
    },
    {
        id: '002',
        rating: 9,
        date: "TEST-DATE",
        text: "Test text"
    },
    {
        id: '003',
        rating: 8,
        date: "TEST-DATE",
        text: "Test text"
    },
    {
        id: '004',
        rating: 7,
        date: "TEST-DATE",
        text: "Test text"
    }
];

export const testLog = {
    id: '001',
    rating: 10,
    timing: 'Afternoon', 
    date: "TEST-DATE",
    text: "Test text"
}