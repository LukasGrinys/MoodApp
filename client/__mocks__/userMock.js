export const testUser = {
    id: '001',
    firstName: 'test',
    lastName: 'test',
    email: 'test@test.com'
}