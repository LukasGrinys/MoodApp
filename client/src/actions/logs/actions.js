export const actions = {
    canLog : {
        request : 'CAN_LOG_REQUEST',
        receive : 'CAN_LOG_RECEIVE'
    },
    getLastLogs : {
        request : 'GET_LAST_LOGS_REQUEST',
        receive : 'GET_LAST_LOGS_RECEIVE',
        reset : 'GET_LAST_LOGS_RESET'
    },
    getLog : {
        request : 'GET_LOG_REQUEST',
        receive : 'GET_LOG_RECEIVE',
        reset : 'GET_LOG_RESET'
    },
    getLogs : {
        request : 'GET_LOGS_REQUEST',
        receive : 'GET_LOGS_RECEIVE',
        reset : 'GET_LOGS_RESET'
    },
    postLog : {
        request : 'POST_LOG_REQUEST',
        receive : 'POST_LOG_RECEIVE',
        reset : 'POST_LOG_RESET'
    }
}