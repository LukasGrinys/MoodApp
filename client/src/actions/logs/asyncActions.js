import axios from 'axios';
import { actions } from './actions';

export const canUserLog = async (currentDate, currentTiming, user) => {
    const body = {
        userId: user,
        date: currentDate,
        timing: currentTiming
    };

    return async dispatch => {
        const { data } = await axios.post('/api/canLog', body);

        dispatch({type: actions.canLog.request})
        
        if (data && data.data) {
            dispatch({
                type: actions.canLog.receive,
                payload: data.data.canLog
            })

            return;
        }

        dispatch({
            type: actions.canLog.receive,
            payload: false
        })
    }
}

export const getLastLogs = async ({
    userId,
    limit = 9
}) => {
    const url = `/api/getLogs?id=${userId}&limit=${limit}`;

    return async dispatch => {
        dispatch({type: actions.getLastLogs.request});

        try {
            const { data } = await axios.get(url);
            const { success, error } = data;
    
            if (error) {
                dispatch({
                    type: actions.getLastLogs.receive,
                    payload: {
                        logs: [],
                        error: error,
                        isFetching: false
                    }
                });
    
                return;
            }
    
            if (success && data.data) {
                dispatch({
                    type: actions.getLastLogs.receive,
                    payload: {
                        logs: data.data,
                        error: null,
                        isFetching: false
                    }
                });
    
                return;
            }
        } catch (error) {
            dispatch({
                type: actions.getLastLogs.receive,
                payload: {
                    logs: [],
                    error: error,
                    isFetching: false
                }
            });
        }
    }
}

export const clearLastLogs = () => dispatch => dispatch({type: actions.getLastLogs.reset});

export const getLog = async logId => {
    return async dispatch => {
        const endpoint = `/api/getLog?id=${logId}`;

        dispatch({type: actions.getLog.request});

        try {
            const response = await axios.get(endpoint);
            
            const { error, data } = response.data;

            if (error) {
                dispatch({
                    type: actions.getLog.receive,
                    payload: {
                        isFetching : false,
                        error,
                        logData : {}
                    }
                });

                return;
            }
            
            dispatch({
                type: actions.getLog.receive,
                payload: {
                    isFetching : false,
                    error : false,
                    logData : data
                }
            });
        } catch (error) {
            dispatch({
                type: actions.getLog.receive,
                payload: {
                    isFetching : false,
                    error,
                    logData : {}
                }
            });
        }
    }
}

export const clearLog = () => dispatch => { dispatch({type: actions.getLog.reset}) }

export const getLogs = async ({userId, start, limit, currentLogs}) => {
    return async dispatch => {
        const endpoint = `/api/getLogs?id=${userId}${start ? `&skip=${start}` : ''}&limit=${limit}`;

        dispatch({ type: actions.getLogs.request });

        try {
            const response = await axios.get(endpoint);
            const { data, error } = response.data;

            if (error) {
                dispatch({
                    type: actions.getLogs.receive,
                    payload: {
                        isFetching: false,
                        error
                    }
                });

                return;
            }
            
            dispatch({
                type: actions.getLogs.receive,
                payload: {
                    isFetching: false,
                    error: false,
                    logs: currentLogs ? [...currentLogs, ...data] : data,
                    noLogsLeft: !data.length ? true : false
                }
            });
        } catch (error) {
            dispatch({
                type: actions.getLogs.receive,
                payload: {
                    isFetching: false,
                    error
                }
            });
        }
    }
}

export const clearLogs = () => dispatch => { dispatch({ type: actions.getLogs.reset }) }

export const postLog = async ({
    userId,
    date,
    timing,
    rating,
    text
}) => {
    const body = {
        userId,
        date,
        timing,
        rating,
        text
    }

    return async dispatch => {
        try {
            dispatch( {type: actions.postLog.request});
    
            const { data } = await axios.post('/api/postLog', body);
            const { error , success } = data;

            if (error || !success) {
                dispatch({
                    type: actions.postLog.receive,
                    payload: {
                        isPosting : false,
                        success : false,
                        error : error || 'Unknown error occured'
                    }
                });

                return;
            }

            dispatch({
                type: actions.postLog.receive,
                payload: {
                    isPosting : false,
                    success : true,
                    error : null
                }
            })
        } catch (error) {
            dispatch({
                type: actions.postLog.receive,
                payload: {
                    isPosting : false,
                    success : false,
                    error
                }
            })
        }
    }  
}

export const clearLogPost = () => dispatch => { dispatch( {type: actions.postLog.reset})};