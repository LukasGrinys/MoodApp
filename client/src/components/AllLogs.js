import React, { Component } from 'react';
import { getLogs } from './../actions';
import { connect } from 'react-redux';

import Loading from './Loading/loading';
import LogItem from './../widgets/logItem';
import styles from './logs.module.css';
import BackButton from './../widgets/backButton';
import ButtonWid from './../widgets/Button';
import LoadingNetItem from './Loading/loadingNetItem';

class LogList extends Component {
    state = {
        skip: 3
    };
    UNSAFE_componentWillMount() {
        this.props.dispatch(getLogs(this.props.user.data.id, 0, 3));
    }
    loadMoreLogs = () => {
        if (this.props.user.data) {
            let skip = this.state.skip;
            this.props.dispatch(getLogs(this.props.user.data.id, skip, 3, this.props.logs.list));
            skip += 3;
            this.setState({
                skip: skip
            });
        }
    };
    returnLogs = (data) => {
        if (!data) return null;

        const list = data.data;
        
        if (list && Array.isArray(list) && list.length) {
            return list.map( (item, i) => {
                return (
                    <LogItem 
                        key={i} 
                        rating={item.rating} 
                        date={item.date} 
                        timing={item.timing} 
                        text={item.text}
                    />
                ) 
            })
        } else {
            return <LoadingNetItem/>
        }
    };
    returnLoadMoreButton = (list) => {
        if (list && list.length > 0) {
            const lastElement = list[list.length - 1];
            if (lastElement !== "No logs found") {
                return (
                    <ButtonWid background={'#3366FF'} color={'#FFFFFF'}>Load more</ButtonWid>
                )
            } else {
                return null
            }
        } else {
            return null;
        }
    }
    render() {
        return (
            <div>
                <BackButton/>
                <div className={styles.header}>All logs</div>
                <div className={styles.logs_wrapper}>
                    {this.returnLogs(this.props.logs.list)}
                </div>
                <div className={styles.bottom_panel}>
                    <div onClick={this.loadMoreLogs}>
                        {this.returnLoadMoreButton(this.props.logs.list)}
                    </div>
                </div>
            </div>
        );
    }
}

class AllLogs extends Component {
    render() {
        if (this.props.user.data) {
            return <LogList {...this.props}/>
        } else {
            return <Loading/>
        }
    }
}

function mapStateToProps(state) {
    return {
        logs: state.logs,
        user: state.user
    }
}
export default connect(mapStateToProps)(AllLogs);