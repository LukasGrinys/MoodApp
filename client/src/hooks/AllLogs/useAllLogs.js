
import { useCallback, useEffect, useState } from 'react';
import { getLogs, clearLogs } from '../../actions';
import { useDispatch, useSelector } from 'react-redux';

export const useAllLogs = () => {
    const dispatch = useDispatch();
    const { user, logs } = useSelector( ({user, logs}) => { 
        return {
            user,
            logs
        }
    });
    const [ skipLogs, setSkipLogs ] = useState(3); 

    const { id : userId } = user;
    const { 
        allLogs, 
        isFetchingAllLogs, 
        fetchAllLogsError,
        noLogsLeft
    } = logs;

    useEffect( () => {
        if (userId) {
            dispatch(getLogs({
                userId, 
                start : 0, 
                limit : 3
            }));
        }
    }, [userId, dispatch]);

    useEffect( () => {
        return () => {
            dispatch(clearLogs());
        }
        // eslint-disable-next-line
    }, [])

    const loadMoreLogs = useCallback( () => {
        if (userId) {
            dispatch(getLogs({
                userId, 
                start : skipLogs, 
                limit : 3,
                currentLogs : allLogs
            }));

            setSkipLogs( prevState => prevState + 3);
        }
    }, [skipLogs, userId, allLogs, dispatch]);

    return {
        allLogs,
        isFetchingAllLogs, 
        fetchAllLogsError,
        noLogsLeft,
        loadMoreLogs
    }
}