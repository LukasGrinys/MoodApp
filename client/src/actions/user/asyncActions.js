import { actions } from './actions';
import axios from 'axios';
import { clearLogs, clearLastLogs } from '../logs/asyncActions';
import Cookie from 'universal-cookie';
const cookie = new Cookie();

export const createUser = async user => {
    return async dispatch => {
        dispatch({type: actions.createUser.request});

        try {
            const { data } = await axios.post('/api/users', user)
            const { success, error } = data;

            if (error || !success) {
                dispatch({
                    type: actions.createUser.receive,
                    payload: {
                        success: false,
                        error: error || 'Could not create user'
                    }
                });

                return;
            }
            
            if (success) {
                dispatch({
                    type: actions.createUser.receive,
                    payload: {
                        success: true,
                        error: null
                    }
                });
            }
        } catch (error) {
            dispatch({
                type: actions.createUser.receive,
                payload: {
                    success: false,
                    error: error || 'Could not create user'
                }
            });
        }
    }
}

export const clearCreateUser = () => dispatch => dispatch({type: actions.createUser.reset});

export const loginUser = async payload => {
    return async dispatch => {
        dispatch({
            type: actions.loginUser.request
        });

        try {
            const { data } = await axios.post('/api/login', payload);
            const { isAuth, error } = data;

            if (error) {
                dispatch({
                    type: actions.loginUser.receive,
                    payload: {
                        isAuth: false,
                        authError : error
                    }
                });

                return;
            }

            if (isAuth) {
                dispatch({
                    type: actions.loginUser.receive,
                    payload: {
                        isAuth,
                        error: null
                    }
                });
            }
        } catch (error) {
            dispatch({
                type: actions.loginUser.receive,
                payload: {
                    isAuth: false,
                    error
                }
            })
        }
    }
}

export function logOut() {
    const request = axios.get('/api/logout');

    return dispatch => {
        request.then( (response) => {
            if (response.status === 200) {
                cookie.remove('auth');
                
                dispatch({
                    type: actions.clearUserData
                });
                
                clearLogs();
                clearLastLogs();
            }
        })
    }
}

export function auth() {
    const response = axios.get('/api/auth')
    .then( ({data}) => data);

    return {
        type: actions.userAuth,
        payload: response
    }
}

export const editUserDetails = async ({userId, details}) => {
    const endpoint = '/api/editAccount';
    const body = {
        id : userId,
        ...details
    }

    return async dispatch => {
        dispatch({type: actions.editUser.request});

        try {
            const { data } = await axios.post(endpoint, body);
            const { error } = data;

            if (error) {
                dispatch({
                    type: actions.editUser.receive,
                    payload: {
                        isEditing: false,
                        success: false,
                        error: error.toString()
                    }
                });

                return
            }

            dispatch({
                type: actions.editUser.receive,
                payload: {
                    isEditing: false,
                    success: true,
                    error: null
                }
            });
        } catch (error) {
            dispatch({
                type: actions.editUser.receive,
                payload: {
                    isEditing: false,
                    success: false,
                    error: error.toString()
                }
            });
        }
    }
}

export const clearEditUser = () => dispatch => {dispatch({type: actions.editUser.reset})}

export const changePassword = async ({
    userId,
    currentPassword,
    newPassword
}) => {
    return async dispatch => {
        if (!userId || !currentPassword || !newPassword) {
            return;
        }

        const endpoint = '/api/changePassword';
        const body = {
            id : userId,
            oldPassword : currentPassword,
            newPassword
        }

        await dispatch({type: actions.changePassword.request});

        try {
            const { error } = await axios.post(endpoint, body);

            if (error) {
                dispatch({
                    type: actions.changePassword.receive,
                    payload: {
                        isChanging : false,
                        error : error,
                        success : false
                    }
                });

                return;
            }

            dispatch({
                type: actions.changePassword.receive,
                payload: {
                    isChanging : false,
                    error : null,
                    success : true
                }
            });
        } catch (error) {
            dispatch({
                type: actions.changePassword.receive,
                payload: {
                    isChanging : false,
                    error : error,
                    success : false
                }
            });
        }
    }
}

export const clearChangePassword = async () => async dispatch => {dispatch({type: actions.changePassword.reset})}

export const clearUserData = async () => async dispatch => {dispatch({type: actions.clearUserData})}