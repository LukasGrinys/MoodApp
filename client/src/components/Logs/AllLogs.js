import React from 'react';
import Loading from '../Loading/loading';
import BackButton from '../Back/BackButton';
import styles from './AllLogs.module.scss';
import Button from '../Button/Button';
import LogItem from '../LogItem/logItem';
import LoadingNetItem from '../Loading/loadingNetItem';
import ErrorBox from '../ErrorBox/ErrorBox';
import { useAllLogs } from '../../hooks/AllLogs/useAllLogs';

const AllLogs = () => {
    const { 
        allLogs,
        isFetchingAllLogs, 
        fetchAllLogsError,
        noLogsLeft,
        loadMoreLogs 
    } = useAllLogs();

    if (!allLogs) {
        return <Loading/>
    }

    if (
        allLogs && 
        Array.isArray && 
        allLogs.length
    ) {
        return (
            <div className={styles.pageWrapper}>
                <BackButton/>
                <h1>All logs</h1>
                <div>
                    {
                        allLogs.map( (item, i) => {
                            return (
                                <LogItem 
                                    key={i} 
                                    rating={item.rating} 
                                    date={item.date} 
                                    timing={item.timing} 
                                    text={item.text}
                                />
                            ) 
                        })
                    }
                </div>
                <div className={styles.bottomSection}>
                    {
                        isFetchingAllLogs ? (
                            <LoadingNetItem/>
                        ) : !noLogsLeft && (
                            <Button
                                handleClick={loadMoreLogs}
                            >
                                Load more
                            </Button>
                        ) 
                    }
                    {
                        fetchAllLogsError && (
                            <ErrorBox>
                                {fetchAllLogsError}
                            </ErrorBox>
                        )
                    }
                </div>
            </div>
        )
    }

}

export default AllLogs;