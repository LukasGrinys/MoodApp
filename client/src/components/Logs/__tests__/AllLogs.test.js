import React from 'react';
import AllLogs from '../AllLogs';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { testLogs } from '../../../../__mocks__/logsMock';
import { getLogs } from '../../../actions/logs/asyncActions';

const mockStore = configureStore();

describe('AllLogs page: loading state', () => {
    const store = mockStore({
        user: {},
        logs: {
            allLogs: {}
        }
    });
    
    const wrapper = mount(
        <Provider store={store}>
            <AllLogs/>
        </Provider>
    );

    test('Renders properly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Has Loading widget', () => {
        expect(wrapper.find('Loading').length).toBe(1);
    });
});

describe('AllLogs page: no logs', () => {
    const store = mockStore({
        user: {},
        logs: {
            allLogs: {
                logs: [], 
                isFetchingAllLogs: false, 
                noLogsLeft: false
            }
        }
    });
    
    const wrapper = mount(
        <Provider store={store}>
            <AllLogs/>
        </Provider>
    );

    test('Renders properly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Has a message "no logs found"', () => {
        expect(wrapper.find('div').at(0).text()).toBe('No logs found');
    })
});

describe('AllLogs page: logs found', () => {
    const store = mockStore({
        user: {},
        logs: {
            allLogs: {
                logs: testLogs, 
                isFetchingAllLogs: false, 
                noLogsLeft: false
            }
        }
    });
    
    const wrapper = mount(
        <Provider store={store}>
            <AllLogs/>
        </Provider>
    );

    test('Renders properly', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    test('It renders LogItem component for each log', () => {
        expect(wrapper.find('LogItem').length).toBe(testLogs.length);
    });

    test('It has button for loading more components', () => {
        expect(wrapper.find('button').length).toBe(1);
    });

    test('It calls getLogs when rendered', () => {
        setTimeout( () => {
            expect(getLogs).toHaveBeenCalledTimes(1);
        }, 0)
    });

    test('It calls getLogs and shows loading widget when clicked on the LoadMore button', () => {
        wrapper.find('button').simulate('click');
        wrapper.update();

        setTimeout( () => {
            expect(wrapper.find('LoadingNetItem').length).toBe(1);
            expect(getLogs).toHaveBeenCalledTimes(2);
        });
    })
});

test('It does not render LoadMore button if all logs have been fetched', () => {
    const store = mockStore({
        user: {},
        logs: {
            allLogs: {
                logs: testLogs, 
                isFetchingAllLogs: false, 
                noLogsLeft: true
            }
        }
    });
    
    const wrapper = mount(
        <Provider store={store}>
            <AllLogs/>
        </Provider>
    );

    expect(wrapper.find('button').length).toBe(0);
});

test('It renders Error Box if there was an error while fetching', () => {
    const testError = 'test error';

    const store = mockStore({
        user: {},
        logs: {
            allLogs: {
                logs: testLogs, 
                isFetchingAllLogs: false, 
                error: testError,
                noLogsLeft: false
            }
        }
    });
    
    const wrapper = mount(
        <Provider store={store}>
            <AllLogs/>
        </Provider>
    );

    expect(wrapper.find('ErrorBox').length).toBe(1);
    expect(wrapper.find('ErrorBox').text()).toBe(testError);
});
