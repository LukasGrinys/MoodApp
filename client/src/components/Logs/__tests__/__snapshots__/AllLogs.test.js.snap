// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllLogs page: loading state Renders properly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <AllLogs>
    <Loading>
      <div
        className="loadingScreen"
      >
        <div>
          Loading
        </div>
        <br />
        <div
          className="barContainer"
        >
          <div
            className="bar"
          />
          <div
            className="bar delay1"
          />
          <div
            className="bar delay2"
          />
        </div>
      </div>
    </Loading>
  </AllLogs>
</Provider>
`;

exports[`AllLogs page: logs found Renders properly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <AllLogs>
    <div
      className="pageWrapper"
    >
      <BackButton>
        <div
          className="container"
        >
          <div
            className="button"
            onClick={[Function]}
          >
            <FontAwesome
              name="arrow-left"
            >
              <span
                aria-hidden={true}
                className="fa fa-arrow-left"
              />
            </FontAwesome>
            <div
              className="text"
            >
              Go back
            </div>
          </div>
        </div>
      </BackButton>
      <h1>
        All logs
      </h1>
      <div>
        <LogItem
          date="TEST-DATE"
          key="0"
          rating={10}
          text="Test text"
        >
          <div
            className="item"
          >
            <div
              className="top_line"
            >
              <div
                className="rating_box"
              >
                10
              </div>
              <div>
                <FontAwesome
                  className="icon"
                  name="hourglass"
                >
                  <span
                    aria-hidden={true}
                    className="fa fa-hourglass icon"
                  />
                </FontAwesome>
              </div>
              <div
                className="date"
              >
                <FontAwesome
                  className="icon"
                  name="calendar"
                >
                  <span
                    aria-hidden={true}
                    className="fa fa-calendar icon"
                  />
                </FontAwesome>
                TEST-DATE
              </div>
            </div>
            <div
              className="text"
            >
              Test text
            </div>
          </div>
        </LogItem>
        <LogItem
          date="TEST-DATE"
          key="1"
          rating={9}
          text="Test text"
        >
          <div
            className="item"
          >
            <div
              className="top_line"
            >
              <div
                className="rating_box"
              >
                9
              </div>
              <div>
                <FontAwesome
                  className="icon"
                  name="hourglass"
                >
                  <span
                    aria-hidden={true}
                    className="fa fa-hourglass icon"
                  />
                </FontAwesome>
              </div>
              <div
                className="date"
              >
                <FontAwesome
                  className="icon"
                  name="calendar"
                >
                  <span
                    aria-hidden={true}
                    className="fa fa-calendar icon"
                  />
                </FontAwesome>
                TEST-DATE
              </div>
            </div>
            <div
              className="text"
            >
              Test text
            </div>
          </div>
        </LogItem>
        <LogItem
          date="TEST-DATE"
          key="2"
          rating={8}
          text="Test text"
        >
          <div
            className="item"
          >
            <div
              className="top_line"
            >
              <div
                className="rating_box"
              >
                8
              </div>
              <div>
                <FontAwesome
                  className="icon"
                  name="hourglass"
                >
                  <span
                    aria-hidden={true}
                    className="fa fa-hourglass icon"
                  />
                </FontAwesome>
              </div>
              <div
                className="date"
              >
                <FontAwesome
                  className="icon"
                  name="calendar"
                >
                  <span
                    aria-hidden={true}
                    className="fa fa-calendar icon"
                  />
                </FontAwesome>
                TEST-DATE
              </div>
            </div>
            <div
              className="text"
            >
              Test text
            </div>
          </div>
        </LogItem>
        <LogItem
          date="TEST-DATE"
          key="3"
          rating={7}
          text="Test text"
        >
          <div
            className="item"
          >
            <div
              className="top_line"
            >
              <div
                className="rating_box"
              >
                7
              </div>
              <div>
                <FontAwesome
                  className="icon"
                  name="hourglass"
                >
                  <span
                    aria-hidden={true}
                    className="fa fa-hourglass icon"
                  />
                </FontAwesome>
              </div>
              <div
                className="date"
              >
                <FontAwesome
                  className="icon"
                  name="calendar"
                >
                  <span
                    aria-hidden={true}
                    className="fa fa-calendar icon"
                  />
                </FontAwesome>
                TEST-DATE
              </div>
            </div>
            <div
              className="text"
            >
              Test text
            </div>
          </div>
        </LogItem>
      </div>
      <div
        className="bottomSection"
      >
        <Button
          handleClick={[Function]}
        >
          <button
            className="button btnPrimary"
            onClick={[Function]}
            type="button"
          >
            Load more
          </button>
        </Button>
      </div>
    </div>
  </AllLogs>
</Provider>
`;

exports[`AllLogs page: no logs Renders properly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <AllLogs>
    <div>
      No logs found
    </div>
  </AllLogs>
</Provider>
`;
